### PR TITLE
support using cached data and re-splitting for huggingface datasets

### PR DESCRIPTION
**Pull request by yxdyc**
support 
- using cached data for huggingface datasets;
-  re-splitting the GLUE versions into custom FL versions

### CODE REVIEW

File: federatedscope/core/auxiliaries/data_builder.py [hunk 1]
```diff
@@ -285,8 +290,14 @@ def load_torchtext_data(name, splits=None, config=None):
 
         if config.model.type.endswith('transformers'):
             from transformers import AutoTokenizer
-            tokenizer = AutoTokenizer.from_pretrained(
-                config.model.type.split('@')[0])
+
+            try:
```
Comment (rayrayraykk) [inline federatedscope/core/auxiliaries/data_builder.py:294]: Why need a try here?

File: federatedscope/core/auxiliaries/data_builder.py [hunk 2]
```diff
@@ -402,6 +413,7 @@ def load_torch_geometric_data(name, splits=None, config=None):
 
     def load_huggingface_datasets_data(name, splits=None, config=None):
         from datasets import load_dataset
+        from datasets import load_from_disk
```
Comment (rayrayraykk) [inline federatedscope/core/auxiliaries/data_builder.py:416]: Merge Line 415 and Line 416.

File: federatedscope/core/auxiliaries/data_builder.py [hunk 3]
```diff
@@ -410,18 +422,46 @@ def load_huggingface_datasets_data(name, splits=None, config=None):
...
-        dataset = load_dataset(path=config.data.root,
-                               name=name,
-                               **filtered_args)
+        logger.info("Begin to load huggingface dataset")
+        if "hg_cache_dir" in raw_args:
+            hugging_face_path = raw_args["hg_cache_dir"]
+        else:
+            hugging_face_path = os.getcwd()
+
+        if "load_disk_dir" in raw_args:
```
Comment (rayrayraykk) [inline federatedscope/core/auxiliaries/data_builder.py:431]: We could use a try here when `load_disk_dir` in `raw_args` but without the data file cached.

**Review by rayrayraykk — COMMENTED** [review]
Good job, we could accelerate to load the huggingface data file with the cache.

File: federatedscope/core/auxiliaries/data_builder.py (hunk 1 above)
Comment (yxdyc) [reply federatedscope/core/auxiliaries/data_builder.py:294]: In case of the cached file not existed

**Review by rayrayraykk — APPROVED** [review]
LGTM

**Pull request merged**